server/instrumentation-backend/src/sh/calaba/instrumentationbackend/actions/text/InfoMethodUtil.java

```diff
@@ -42,23 +40,62 @@ static InputConnection getInputConnection() {
             servedInputConnectionField.setAccessible(true);
 
             return (InputConnection) servedInputConnectionField.get(inputMethodManager);
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-    static InputConnection tryGetInputConnection() {
-        try {
-            return getInputConnection();
-        } catch (Exception e) {
-            return null;
+        } catch (IllegalAccessException e) {
+            throw new UnexpectedInputMethodManagerStructureException(e);
+        } catch (NoSuchFieldException e) {
+            throw new UnexpectedInputMethodManagerStructureException(e);
         }
     }
 
     /*
      Find length of non-formatted text
     */
-    static int getEditableTextLength(Editable editable) {
+    public static int getEditableTextLength(Editable editable) {
         return TextUtils.substring(editable, 0, editable.length()).length();
     }
+
+    public static Editable getEditable(View view) {
+        Editable editable = null;
+
+        if (view instanceof TextView) {
+            editable = ((TextView) view).getEditableText();
+        } else {
+            try {
+                Method m = view.getClass().getMethod("getEditableText");
```

Comment:
You are right, but the fields could potentially include a lot of editable references to non-relevant stuff.